src/spec.c

```diff
@@ -1148,20 +1160,41 @@ void IndexSpec_FreeSync(IndexSpec *spec) {
 
 //---------------------------------------------------------------------------------------------
 
-void Indexes_Free(dict *d) {
-  arrayof(IndexSpec *) specs = array_new(IndexSpec *, 10);
-  dictIterator *iter = dictGetIterator(d);
+void Indexes_Free(dict *specsDict, TrieMap *schemaPrefixes, void *aliases, void *cursors) {
+  if (!cleanPool) {
+    cleanPool = thpool_init(1);
+  }
+
+  arrayof(IndexSpec *) specs = array_new(IndexSpec *, dictSize(specsDict));
+  dictIterator *iter = dictGetIterator(specsDict);
   dictEntry *entry = NULL;
+
+  // save all specs to an array and empty global structures (unlink):
+  // specs, schema prefixes, aliases, cursors, and GC 
   while ((entry = dictNext(iter))) {
     IndexSpec *sp = dictGetVal(entry);
     specs = array_append(specs, sp);
+
+    if (sp->isTimerSet) {
+      RedisModule_StopTimer(RSDummyContext, sp->timerId, NULL);
+      sp->isTimerSet = false;
+    }
+    if (cursors && sp->uniqueId) {
+      Cursors_PurgeWithName(cursors, sp->name);
+      CursorList_RemoveSpec(cursors, sp->name);
+    }
+    if (sp->gc) {
+      GCContext_Stop(sp->gc);
+    }
```

Comment:
I understand we are not calling IndexSpec_FreeGlobals here on purpose (its easier to clean the global structures), let document the function and explain it.